src/SAML2/XML/saml/EncryptedAssertion.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML\saml;
+
+use RobRichards\XMLSecLibs\XMLSecurityKey;
+use SAML2\EncryptedElementInterface;
+use SAML2\EncryptedElementTrait;
+use SAML2\Utils;
+use SAML2\XML\AbstractXMLElement;
+
+/**
+ * Class handling encrypted assertions.
+ *
+ * @package SimpleSAMLphp
+ */
+class EncryptedAssertion extends AbstractSamlElement implements EncryptedElementInterface
+{
+    use EncryptedElementTrait;
+
+    /**
+     * @var bool
+     */
+    protected $wasSignedAtConstruction = false;
+
+
+    /**
+     * @inheritDoc
+     *
+     * @return \SAML2\XML\saml\Assertion
+     * @throws \Exception
+     *
+     * @psalm-suppress MismatchingDocblockReturnType
+     * @psalm-suppress ImplementedReturnTypeMismatch
+     */
+    public function decrypt(XMLSecurityKey $key, array $blacklist = []): AbstractXMLElement
```

Comment:
We may want to avoid the @psalm-suppress by following the same as we did here using the object-return type: https://github.com/simplesamlphp/saml2/blob/6c04f44afd3f8b711cdbb77c964817094bb55359/src/SAML2/XML/AbstractXMLElement.php#L246

Reply:
The problem here is that while `fromXML()` returns an object of the same class, `decrypt()` does not. So `self` is not an option, unfortunately (unless I'm misunderstanding your suggestion)...

Reply:
I never said anything about `self`... I'm talking about using 'object' as a return type and specify the specific type in the phpdoc...

Reply:
Using `object` was a mistake, in my opinion. It doesn't give you anything, it doesn't offer any API you are guaranteed to have, even though elements created with `fromXML()` are always supposed to be descendants of `AbstractXMLElement`. At least it should have been the latter.